src/components/pages/farmer-member/map/geo-json-map.tsx

```diff
@@ -0,0 +1,106 @@
+import L from "leaflet";
+import React, { useEffect, useMemo } from "react";
+import { GeoJSON, LayerGroup, LayersControl, MapContainer, TileLayer, useMap } from "react-leaflet";
+
+const GeoJsonMap = ({ geoJsonData }) => {
+  // Calculate center based on GeoJSON data
+  const center = useMemo(() => {
+    if (geoJsonData && geoJsonData.type === "Feature" && geoJsonData.geometry.type === "Point") {
+      // Directly return the coordinates for a single point
+      return [geoJsonData.geometry.coordinates[1], geoJsonData.geometry.coordinates[0]];
+    } else if (
+      geoJsonData &&
+      geoJsonData.type === "Feature" &&
+      geoJsonData.geometry.type === "MultiPoint"
+    ) {
+      // Calculate the average of coordinates for a MultiPoint
+      const coords = geoJsonData.geometry.coordinates;
+      const average = coords.reduce(
+        (acc, coord) => {
+          acc[0] += coord[0] / coords.length;
+          acc[1] += coord[1] / coords.length;
+          return acc;
+        },
+        [0, 0]
+      );
+      return [average[1], average[0]]; // Convert [long, lat] to [lat, long] for Leaflet
+    }
+    // Default center if no data or unsupported type
+    return [42.35725, -71.060982];
```

Comment:
if there is a special relevance of these coordinates , extract them in meaningful constants.